flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2116,8 +2171,34 @@ struct StoreOpConversion : public FIROpConversion<fir::StoreOp> {
   mlir::LogicalResult
   matchAndRewrite(fir::StoreOp store, OperandTy operands,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    rewriter.replaceOpWithNewOp<mlir::LLVM::StoreOp>(store, operands[0],
-                                                     operands[1]);
+    if (store.value().getType().isa<fir::BoxType>()) {
+      // The box value is in memory, load it first before storing it into
+      // the destination address.
```

Comment:
This code is actually doing a memcpy instruction.

Reply:
Thanks outdated comments.
I first try copying descriptor with 
```
  %0 = llvm.load %desc_src_addr
  llvm.store %0 to %desc_dest_addr
```

But llvm was not happy about loading/storing these llvm.struct types. I need to dig more because I think it should be OK (and if we want to make box real ssa values, it better works, otherwise loading/storing boxes into memory will be a painful succession of GEP + load (or store) and insert (or extract) field by field).

Reply:
Yep. Definitely would like LLVM to understand it. ;)

Reply:
I may have used the wrong type in my first attempt, that of course works. I rolled back the memcpy that was more complex to use.